include/NAS2D/Random.h

```diff
@@ -0,0 +1,196 @@
+#pragma once
+
+#include <random>
+#include <type_traits>
+
+#if defined(_WIN64) || defined(__MINGW64__)
+using Rng_t = std::mt19937_64;
+#else
+using Rng_t = std::mt19937;
+#endif
+
+namespace NAS2D
+{
+	namespace MathUtils
+	{
+		class Random
+		{
+		  public:
+			Random(const Random&) = delete;
+			Random& operator=(const Random&) = delete;
+			Random(Random&&) = delete;
+			Random& operator=(Random&&) = delete;
+
+			/**
+			* \fn static void seed(std::size_t seedValue)
+			* 
+			* Seeds the random number generator with a specific value.
+			* 
+			* \param seedValue: The value with which to seed the generator.
+			* \return Nothing
+			* \remarks
+			* If not called prior to any other function, then the RNG is seeded with a non-deterministic random value.
+			*/
+			static void seed(std::size_t seedValue) noexcept
```

Comment:
It wasn't. The basis was the user story: "As a user, I want to be able to re-seed the generator with a value read from a file."

E.g. saving generated seeds inside save files to be able to reproduce "random" values in a deterministic fashion.

Reply:
That works for game replay, assuming both are initialized with the same limited seed. It has different implications for saved games, since it is insufficient to save and restore the entire state of the random number generator. That may be fine, if you want to allow for a new random sequence each time you load. Maybe not.

Reply:
Huh?